x-pack/plugins/uptime/public/components/monitor_management/monitor_list/monitor_async_error.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState } from 'react';
+import { useSelector } from 'react-redux';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiButton, EuiCallOut, EuiSpacer } from '@elastic/eui';
+import { monitorManagementListSelector } from '../../../state/selectors';
+
+export const MonitorAsyncError = () => {
+  const [isDismissed, setIsDismissed] = useState(false);
+  const { list, locations } = useSelector(monitorManagementListSelector);
+  const syncErrors = list.syncErrors;
+  const hasSyncErrors = syncErrors && syncErrors.length > 0;
+
+  return hasSyncErrors && !isDismissed ? (
+    <>
+      <EuiCallOut
+        title={
+          <FormattedMessage
+            id="xpack.uptime.monitorManagement.monitorSync.failure.title"
+            defaultMessage="Monitors failed to sync with the Synthetics service"
+          />
+        }
+        color="warning"
+        iconType="alert"
+      >
+        <p>
+          <FormattedMessage
+            id="xpack.uptime.monitorManagement.monitorSync.failure.content"
+            defaultMessage="There was a problem syncing your monitors for one or more locations:"
+          />
+        </p>
+        <ul>
+          {Object.values(syncErrors).map((e) => {
+            return (
+              <li key={e.locationId}>{`${
+                locations.find((location) => location.id === e.locationId)?.label
```

Comment:
This ended up being undefined when I tested E2E as no matching locations were found.

Reply:
> LGTM, only small nits.
> 
> Also tested E2E and was able to see the warning after changing the auth so that I can't auth with the service anymore, as shown in the screenshot below.
> 
> ![Screenshot 2022-03-29 at 21 43 55](https://user-images.githubusercontent.com/6868147/160703905-3e8fd335-596e-4235-b88b-7575b9025913.png)
> 
> ⚠️ My only concern is that in my test the warning says `undefined` in the box on the top. Considering FF is today I think we can get this in and fix the location name bug post FF, that's why I'm marking this as approved. Otherwise, I'd "request changes".
> 
> I'll defer to you on whether you think this should be blocking the merge.
> 
> > Please notice I tested against the remote dev service using `ngrok` as instructed in the PR.
> 
> PS: Didn't know about the configurable interval, that's great!

The location undefined issues happens when locations are not loaded beforehand. I will block this screen from rendering until locations is loaded. I thought that was implemented correctly as preventing the screen from rendering until locations was loaded should have been addressed in a separate PR that was merged a few hours ago. However, there may have been an issue with merge conflicts after merging upstream.